### DISCUSSION TITLE
[Networking Noob] Lokinet WAN/LAN Router Support?

### DISCUSSION THREAD

**Discussion by ThisNekoGuy** [oxen-io/lokinet#2056] [Q&A]
I'm not exactly savvy with networking at all; could someone explain to me if and how one would be able to rout all home router traffic through Lokinet?

**Comment by majestrate** [comment]
yes you can as it is the nicest way to use lokinet and is the use case the entire protocol was designed around. we have a docs page in the repo for it. see https://github.com/oxen-io/lokinet/blob/dev/docs/ideal-ux.md#can-i-run-lokinet-on-a-soho-router

**Reply by ThisNekoGuy** [reply]
I see that only seems to support Raspberry Pi, which is fine but I don't have one right now; if I have a spare laptop sitting around... how could I have it do that, if I may ask?

**Reply by majestrate** [reply]
in theory as long as it is a debian linux system with 1 wireless and 1 wired network interface, it should work.

**Reply by ThisNekoGuy** [reply]
Starting to think I'm too stupid to understand how controlling Linux DNS routing (or just networking in general) works Lol
I've been trying to understand this all day on my Arch system...

My system uses `NetworkManager` and I'm struggling to understand how to properly configure forcing Lokinet DNS lookups
I get this is kind of a "me" problem, since I use Arch and not a "redistributable" distro like Debian, but if we had a "setup for dummies" guide in the repo, that would be great Lol

**Reply by majestrate** [reply]
there are high level requirements for that in the "the fun way" section: https://github.com/oxen-io/lokinet/blob/dev/docs/ideal-ux.md#the-fun-way-diy but the instructions for that route will always depend on the distro and hardware itself.

**Reply by ThisNekoGuy** [reply]
I read that but I didn't quite know what to do with that information given that I'm not necessarily experienced on this subject; I understand that there's configuration files but I have next to no understanding on how to properly configure them.

With that said, [I've created a stub article for this topic on the Arch Wiki](https://wiki.archlinux.org/title/Lokinet) for anyone willing to provide more detailed instructions for helping ignorant users such as myself properly set things up.
Linking to this here in case someone else sees this discussion and is willing to contribute to the article.

**EDIT:**
The Arch Wiki admin(s) are trying to remove the wiki page I mentioned because it didn't include user documentation from the start, so...
[Here's a web archive link, for if that happens before it gets updated.](https://web.archive.org/web/20221130011900/https://wiki.archlinux.org/title/Lokinet)
